test/integration/include/seperate.test.js

```diff
@@ -61,6 +61,44 @@ if (current.dialect.supports.groupedLimit) {
         });
       });
 
+      it('should run a belongsToMany association in a separate query', function () {
+        var User = this.sequelize.define('User', {})
+          , Group = this.sequelize.define('Group', {})
+          , sqlSpy = sinon.spy();
+
+        User.Groups = User.belongsToMany(Group, {as: 'groups', through: 'UserGroup'});
+
+        return this.sequelize.sync({force: true}).then(function () {
+          return Promise.join(
+            User.bulkCreate([1, 2].map(function(id) {
+              return {id: id};
+            })),
+            Group.bulkCreate([1, 2].map(function(id) {
+              return {id: id};
+            }))
+          ).then(_.spread(function (users, groups) {
+            return Promise.join(
+              users[0].addGroup(groups[0]),
+              users[1].addGroups(groups)
+            );
+          })).then(function () {
+            return User.findAll({
+              include: [
+                {association: User.Groups, separate: true}
+              ],
+              logging: sqlSpy
+            });
+          }).then(function (users) {
+            expect(users[0].groups).to.be.ok;
+            expect(users[0].groups.length).to.equal(1);
+            expect(users[1].groups).to.be.ok;
+            expect(users[1].groups.length).to.equal(2);
+            expect(users[1].groups).to.include(users[0].groups[0]);
+            expect(sqlSpy).to.have.been.calledThrice;
+          });
+        });
+      });
```

Comment:
I haven't forgotten about it, but what I'm saying still applies-
if we do a 1:M on the through->target, a group with 100 users will be duplicated on 100 rows.
Since this is basically a `hasMany` association between the through to the target, and the user requested a `separate: true` query, then I think the `hasMany` association should also behave with `separate: true`. BTW, this gives me the interesting idea of calling `hasMany.get` from `belongsToMany.get`.

Am I missing something?